components/formats-bsd/src/loci/formats/codec/ZstdCodec.java

```diff
@@ -1,46 +1,39 @@
- *
- * @author Wim Pomp w.pomp at nki.nl
+/*
+ * #%L
+ * BSD implementations of Bio-Formats readers and writers
+ * %%
+ * Copyright (C) 2005 - 2022 Open Microscopy Environment:
+ *   - Board of Regents of the University of Wisconsin-Madison
+ *   - Glencoe Software, Inc.
+ *   - University of Dundee
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
  */
-public class ZstdCodec extends BaseCodec {
 
-    @Override
-    public byte[] compress(byte[] data, CodecOptions options)
-        throws FormatException
-    {
-        if (data == null || data.length == 0)
-            throw new IllegalArgumentException("No data to compress");
-        // TODO: Add compression support.
-        throw new UnsupportedCompressionException("Zstandard Compression not currently supported.");
-    }
-
-    @Override
-    public byte[] decompress(RandomAccessInputStream in, CodecOptions options)
-        throws FormatException, IOException
-    {
-        ByteVector bytes = new ByteVector();
-        byte[] buf = new byte[8192];
-        int r;
-        // read until eof reached
-        try {
-            while ((r = in.read(buf, 0, buf.length)) > 0) bytes.add(buf, 0, r);
-        }
-        catch (EOFException ignored) { }
+package loci.formats.codec;
 
-        byte[] data = bytes.toByteArray();
-        ZstdDecompressor decompressor = new ZstdDecompressor();
-        byte[] output = new byte[(int) ZstdDecompressor.getDecompressedSize(data, 0, data.length)];
-        decompressor.decompress(data, 0, data.length, output, 0, output.length);
-        return output;
-    }
-}
+public class ZstdCodec extends WrappedCodec {
+  public ZstdCodec() {
+    super(new ome.codecs.ZstdCodec());
+  }
+}
```

Comment:
The wrapper needs to explicitly wrap any public API methods in `ome.codecs.ZstdCodec` which aren't part of the `Codec` interface.